tests/mockserver.py

```diff
@@ -117,6 +117,14 @@ def render_POST(self, request):
                 "price": "10",
                 "currency": "USD",
             }
+            extract_from = request_data.get("productOptions", {}).get("extractFrom")
+            if extract_from:
+                from scrapy_zyte_api.providers import ExtractFrom
+
+                if extract_from == ExtractFrom.httpResponseBody:
+                    assert isinstance(response_data["product"], dict)
+                    assert isinstance(response_data["product"]["name"], str)
```

Comment:
These are needed because of how `response_data` is typed, otherwise mypy thinks the values could be anything supported by JSON.